proto/observer/query.proto

```diff
@@ -79,6 +81,34 @@ service Query {
   rpc GetAllBlameRecords(QueryAllBlameRecordsRequest) returns (QueryAllBlameRecordsResponse) {
     option (google.api.http).get = "/zeta-chain/observer/get_all_blame_records";
   }
+
+  rpc GetAllBlockHeaders(QueryAllBlockHeaderRequest) returns (QueryAllBlockHeaderResponse) {
+    option (google.api.http).get = "/zeta-chain/observer/get_all_block_headers";
+  }
+
+  rpc GetBlockHeaderByHash(QueryGetBlockHeaderByHashRequest) returns (QueryGetBlockHeaderByHashResponse) {
+    option (google.api.http).get = "/zeta-chain/observer/get_block_header_by_hash/{block_hash}";
+  }
+
+  // merkle proof verification
+  rpc Prove(QueryProveRequest) returns (QueryProveResponse) {
+    option (google.api.http).get = "/zeta-chain/observer/prove";
+  }
+}
+
+message QueryProveRequest {
+  uint64 chain_id = 1;
+  string tx_hash = 2;
+  ethereum.Proof proof = 3;
+  string block_hash = 4;
+  int64 tx_index = 5;
+  ethereum.Proof proof = 5;
```

Comment:
Sorry, my code suggestion messed up this message. You'll have to delete repeated fields 5 6 7, Lol

Reply:
done